### PR TITLE
improve shutdown of TaskRunner, #21430

### PR DESCRIPTION
**Pull request by patriknw**
- important that no tasks are run after shutdown
- could be a reason for segfault

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #21430

File: akka-remote/src/main/scala/akka/remote/artery/TaskRunner.scala [hunk 1]
```diff
@@ -15,6 +15,7 @@ import scala.annotation.tailrec
 import scala.concurrent.{ Future, Promise }
 import scala.reflect.ClassTag
 import scala.util.control.NonFatal
+import java.util.Arrays
```
Comment (johanandren) [inline akka-remote/src/main/scala/akka/remote/artery/TaskRunner.scala]: Not used anywhere?

**Review by johanandren — APPROVED** [review]
LGTM!

File: akka-remote/src/main/scala/akka/remote/artery/TaskRunner.scala (hunk 1 above)
Comment (patriknw) [inline akka-remote/src/main/scala/akka/remote/artery/TaskRunner.scala]: good catch, failed first attempt

**Comment by drewhk** [comment]
LGTM

**Pull request merged**